src/hotspot/cpu/s390/upcallLinker_s390.cpp

```diff
@@ -66,9 +67,10 @@ static void preserve_callee_saved_registers(MacroAssembler* _masm, const ABIDesc
   __ block_comment("{ preserve_callee_saved_regs ");
   for (int i = 0; i < Register::number_of_registers; i++) {
     Register reg = as_Register(i);
-    // Z_SP and Z_R14 saved/restored by prologue/epilogue
-    if (reg == Z_SP || reg == Z_R14) continue;
-    if (!abi.is_volatile_reg(reg)) {
+    // Z_SP saved/restored by prologue/epilogue
+    if (reg == Z_SP) continue;
+    // though Z_R6 is argument register it is a saved register
```

Comment:
Suggestion:
`    // although Z_R6 is used for parameter passing, it must be saved and restored by a called function.`

Reply:
Thanks for the review. Fixed.